src/account/default.ts

```diff
@@ -116,6 +117,56 @@ export class Account extends Provider implements AccountInterface {
     };
   }
 
+  public async estimateAccountDeployFee(
+    {
+      classHash,
+      addressSalt = 0,
+      constructorCalldata = [],
+      contractAddress: providedContractAddress,
+    }: DeployAccountContractPayload,
+    { blockIdentifier, nonce: providedNonce }: EstimateFeeDetails = {}
+  ): Promise<EstimateFee> {
+    const nonce = toBN(providedNonce ?? (await this.getNonce()));
+    const version = toBN(transactionVersion);
+    const chainId = await this.getChainId();
+
+    let contractAddress = ZERO.toString();
+
+    if (!providedContractAddress) {
```

Comment:
what do you think about extracting this one in a separate method, since we have this also in `deployAccount` ?